Sources/OpenTelemetrySdk/Metrics/Stable/Aggregation/AdaptingCircularBufferCounter.swift

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+class AdaptingCircularBufferCounter: NSCopying {
+    func copy(with zone: NSZone? = nil) -> Any {
+        let copy = AdaptingCircularBufferCounter(maxSize: maxSize)
+        copy.startIndex = startIndex
+        copy.endIndex = endIndex
+        copy.baseIndex = baseIndex
+        copy.backing = backing.copy() as! AdaptingIntegerArray
+        return copy
+    }
+
+    private static let NULL_INDEX: Int = Int.min
```

Comment:
Please use Swift style names even  for constants (no capitals and no underscore even for constants. If you want something more visual maybe you can wrap it in an enum like:
```
 enum Index {
 case null = -1
}
```
And use it like `public private(set) var endIndex = Index.null`